src/x/serialize/encode_decode_prop_test.go

```diff
@@ -90,6 +90,25 @@ func TestPropertyAnyReasonableTagSlicesAreAight(t *testing.T) {
 	properties.TestingRun(t)
 }
 
+func TestPropertyTagsWithMaximumLengthLiteralsAreFine(t *testing.T) {
+	params := gopter.DefaultTestParameters()
+	params.MinSuccessfulTests = 5
+	properties := gopter.NewProperties(params)
+	properties.Property("tags with maximum length literals are handled fine", prop.ForAllNoShrink(
+		func(tags ident.Tags) (bool, error) {
+			iter := ident.NewTagsIterator(tags)
+			newIter, err := encodeAndDecode(iter)
+			if err != nil {
+				return false, err
+			}
+			return tagItersAreEqual(iter, newIter)
+		},
+		maximumLiteralLengthTags().WithLabel("input tags"),
+	))
+
+	properties.TestingRun(t)
+}
```

Comment:
Somehow I think that's an overkill.

Reply:
Removed https://github.com/m3db/m3/pull/3815/commits/47395508bc7242a50f21e12c82cc21a72c846204